sirepo/job_supervisor.py

```diff
@@ -509,7 +522,13 @@ def _ops_to_cancel():
         ):
             # Valid, completed, transient simulation
             # Read this first https://github.com/radiasoft/sirepo/issues/2007
-            return await self._receive_api_runStatus(req)
+            r = await self._receive_api_runStatus(req)
+            if r.state == job.MISSING:
+                # happens when the run dir is deleted w/o us knowing (ex pkcli.admin.purge_free_users)
+                assert not called_from_self, \
```

Comment:
`called_from_self` is a bit confusing. `self` is a meaningful concept, and it's a parameter on the call. Rather have it be a counter: recursion_depth, recursion_level, call_depth, etc. When the counter exceeds a threshold, it should throw an error. In this case, `1` is an appropriate threshold.